perun-core/src/main/java/cz/metacentrum/perun/core/entry/UsersManagerEntry.java

```diff
@@ -1148,6 +1146,37 @@ public void deletePassword(PerunSession sess, String userLogin, String loginName
 		}
 	}
 
+	@Override
+	public void deletePassword(PerunSession sess, User user, String loginNamespace) throws
+			PrivilegeException, UserNotExistsException, LoginNotExistsException, InvalidLoginException, PasswordOperationTimeoutException, PasswordDeletionFailedException {
+		Utils.checkPerunSession(sess);
+
+		// Authorization
+		if(!AuthzResolver.authorizedInternal(sess, "deletePassword_User_String_policy", user)
+			&& (!(AuthzResolver.authorizedInternal(sess, "service_user-deletePassword_User_String_policy", user)) && user.isServiceUser())) {
+			throw new PrivilegeException(sess, "deletePassword");
+		}
```

Comment:
This is a pretty complex condition. It is a bit confusing to understand. I also think there are two problems with it: for the serviceUser, also the first authorizedInternal is evaluated. This way, the service_user policy cannot be more restrictive.

Second, I think there are wrong brackets. The second negation is now only used for the authorizedInternal call. This way, I think that all users, which are not privileged to call this method and also are not service users, would be allowed.

I think this whole thing should be simplified like this:
```suggestion
		boolean authorized;
		if (!user.isServiceUser()) {
			authorized = AuthzResolver.authorizedInternal(sess, "deletePassword_User_String_policy", user);
		} else {
			authorized = AuthzResolver.authorizedInternal(sess, "service_user-deletePassword_User_String_policy", user);
		}
		if(!authorized) {
			throw new PrivilegeException(sess, "deletePassword");
		}
```

Reply:
I put this format to all the methods that used it the same way - `reserveRandomPassword`, `reservePassword`, `validatePassword`, `deletePassword` and `loginExist`.